priority/unix.go

```diff
@@ -10,20 +11,31 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+const selfPID = 0
+
 // SetNice sets the unix "nice" value of the current process
 func SetNice(priority int) error {
 	var err error
-	// pid 0 means "self"
-	pid := 0
 
 	if priority < -20 || priority > 19 {
 		return fmt.Errorf("unexpected priority value %d", priority)
 	}
 
-	clog.Debugf("setting process priority to %d", priority)
-	err = unix.Setpriority(unix.PRIO_PROCESS, pid, priority)
+	// Move ourselves to a new process group so that we can use the process
+	// group variants of Setpriority to affect all of our processes at once
+	err = unix.Setpgid(selfPID, 0)
 	if err != nil {
-		return fmt.Errorf("error setting process priority: %v", err)
+		return fmt.Errorf("cannot set process group, restic will run with the default priority: %w", err)
+	}
+
+	clog.Debugf("setting group process priority to %d", priority)
```

Comment:
Note that this also says "group process" and is not fixed by bf118a0887f158b1126ec385a298fac4db2c4d92.

Sorry, I thought that "same remarks as above" in my other comment would make it clear that _both_ my remarks (the typo and the fallback behavior) apply; that was apparently not clear enough, my bad.